### PR TITLE
remove redundant compiled imdocker-unit-tests 

### PR DESCRIPTION
**Pull request by n2yen**

### CODE REVIEW

**Comment by n2yen** [comment]
#3596

**Comment by n2yen** [comment]
Hi @rgerhards, on a few of the docker platforms, the imdocker tests are failing with 'docker: command not found' errors. I'm not sure i've seen this error before.
Looks like these:
- 'rsyslog debian9 rsyslog'
- 'rsyslog docker-arm-ubuntu18'
- 'rsyslog docker-ubuntu18-san-rsyslog'
- 'rsyslog docker-ubuntu18-distcheck rsyslog'
perhaps a few others - should i change tests to use a specific path when running 'docker'?

**Comment by rgerhards** [comment]
They shouldn't run on these platforms because the are inside a container. Did you change configure settings?

**Comment by n2yen** [comment]
yes, you're right. i removed 'enable_imdocker_tests', which i had originally used to a macro like 'ENABLE_IMDOCKER_UNIT_TESTS'...

**Comment by rgerhards** [comment]
Yes, we definitely need to control this.

**Comment by n2yen** [comment]
ok wil do.

**Comment by rgerhards** [comment]
Being on the PC again. Just for clarity: we need to have the ability to control the run of tests because they can only run on platforms that have docker available. This is NOT the case for the majority of test systems (neither in our CI nor on users systems). So we could either check in each test and skip or have the configure option. I prefer the later because many SKIPS clutter the testbench log.

File: contrib/imdocker/imdocker.c [hunk 1]
```diff
@@ -64,7 +64,6 @@ extern int Debug;
 
 #define USE_MULTI_LINE
 //#define ENABLE_DEBUG_BYTE_BUFFER
```
Comment (rgerhards) [inline contrib/imdocker/imdocker.c]: pls remove commented out code - you could change to #define ENABL... 0/1 and then check that value

File: contrib/imdocker/imdocker.c [hunk 2]
```diff
@@ -1378,15 +1362,13 @@ imdocker_container_logs_curlCB(void *data, size_t size, size_t nmemb, void *buff
...
-			DBGPRINTF("%s() - write size is same as payload_size\n", __FUNCTION__);
-			/* NOTE: We do see if a log line gets extended beyond 16K
-			 * if (mem->data[mem->len-1] == imdocker_eol_char)
-			 */
-			req->submitMsg(pInst, pDataBuf, (const uchar*)DOCKER_TAG_NAME);
-		}
+	CHKiRet(dockerContLogsBufWrite(pDataBuf, pread, write_size));
+	if (pDataBuf->bytes_remaining == 0) {
+		DBGPRINTF("%s() - write size is same as payload_size\n", __FUNCTION__);
+		/* NOTE: We do see if a log line gets extended beyond 16K
```
Comment (rgerhards) [inline contrib/imdocker/imdocker.c]: change to not use commented out code

File: tests/travis/install.sh [hunk 3]
```diff
@@ -83,7 +83,7 @@ else
 	if [ "x$IMDOKER" == "xYES" ]; then
 #    git clone https://github.com/curl/curl.git > /dev/null
 #    (cd curl; ./buildconf; ./configure; make -j2 > /dev/null; sudo make install > /dev/null;)
-		export IMDOCKER_OPT="--enable-imdocker --enable-imdocker-tests";
+		export IMDOCKER_OPT="--enable-imdocker";
```
Comment (rgerhards) [inline tests/travis/install.sh]: on Travis we have docker, so we can (and should) use the tests here.

File: tests/travis/run-osx.sh [hunk 4]
```diff
@@ -36,7 +36,7 @@ autoreconf --force --verbose --install
 export CONFIG_FLAGS="--prefix=/opt/rsyslog --enable-silent-rules --enable-testbench --enable-imbatchreport --enable-imdiag --enable-imfile --enable-improg --enable-impstats --enable-mmrm1stspace --enable-mmanon --enable-mmaudit --enable-mmfields --enable-mmjsonparse --enable-mmpstrucdata --enable-mmsequence --enable-mmutf8fix --enable-mail --enable-omprog --enable-omstdout --enable-omuxsock --enable-pmaixforwardedfrom --enable-pmciscoios --enable-pmcisconames --enable-pmlastmsg --enable-pmsnare --enable-mmcount --disable-omudpspoof --enable-snmp --enable-mmsnmptrapd --disable-uuid --disable-libgcrypt \
 	--enable-pmnull \
 	--enable-pmnormalize=no \
-	--enable-imdocker --enable-imdocker-tests \
+	--enable-imdocker \
```
Comment (rgerhards) [inline tests/travis/run-osx.sh:39]: right! on OSx we do not have docker at all.

File: tests/travis/run.sh [hunk 5]
```diff
@@ -35,7 +35,7 @@ if [ "$MERGE" == "YES" ]; then
 fi
 
 if [ "$IMDOCKER" == "YES" ]; then
-	export IMDOCKER_OPT="--enable-imdocker --enable-imdocker-tests"
+	export IMDOCKER_OPT="--enable-imdocker"
```
Comment (rgerhards) [inline tests/travis/run.sh]: that's for linux, so here the tests should run on travis

**Comment by rgerhards** [comment]
@n2yen fyi: I have created https://discuss.lgtm.com/t/commented-out-c-code-not-detected/1998 because LGTM.com should have detected the commented out code. Just stumbled over it by "accident" ;-)

**Comment by n2yen** [comment]
sounds good. I mistakenly removed the tests - putting them back as you advised, sorry about that!

**Comment by rgerhards** [comment]
@n2yen you can ignore the PR structure validation fail - in this case it's justified

**Comment by rgerhards** [comment]
side-note: nevertheless, that's why I suggest to do one thing per PR. In that case, you could simply have amended to the first commit.

**Comment by n2yen** [comment]
yup, i will rebase them into single commit.

**Comment by rgerhards** [comment]
It's actually 2 things, so two commits is perfect. I was just suggesting 2 prs with one commit each.

**Comment by rgerhards** [comment]
FYI: https://rainer.gerhards.net/2019/03/howto-great-pull-request.html

**Comment by n2yen** [comment]
really great post. I can separate the two commits for clarity. i was hesitant of creating another PR because of possible merge conflicts later. Is it ok to keep both in this PR?

**Comment by rgerhards** [comment]
guidelines are good to have, but they are only guidelines. I see two commits here in one PR as appropriate. Will probably merge later today (off to a meeting soon).

**Pull request merged**